### PR TITLE
Update pylint config to include valid short names

### PR DESCRIPTION
**Pull request by justinchuby**
### Description
Update pylint config to include valid short names
Also disabled `too-many-arguments` and `too-many-locals`

### Motivation and Context
Refine config to reduce lint noise

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -16,13 +16,21 @@ extend_skip_glob = [
 [tool.pydocstyle]
 convention = "google"
 
+[tool.pylint.BASIC]
+good-names = [
+    "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n",
```
Comment (edgchen1) [inline pyproject.toml:21]: why is "o" not included?

**Review by edgchen1 — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (justinchuby) [reply pyproject.toml:21]: Good question :D Are there examples when `o` is used as a good name? I couldn't think of any when I was compiling the list

Comment (edgchen1) [reply pyproject.toml:21]: I don't use one letter variable names very often. it's just that every other letter is there, so I was wondering about the omission.

Comment (justinchuby) [reply pyproject.toml:21]: I could have added a comment but the ci would then have to run for another half day. I will merge this and add some explanation when I work on linting later.

**Pull request merged**